# 胆道闭锁数据集GSE122340的微环境分析

## 1.下载GSE122340数据集，上游分析

```{bash eval =F}
#1.1 进入ENA数据库，下载BioProject：PRJNA504659（BA：171；normal controls:7---228个测序文件），得到ena-file-download-20230616-0255.sh

sed 's/wget -nc/axel -n 20/g' ena-file-download-20230616-0255.sh> download.sh

bash download.sh

##下载完成后，未发现有不完整的下载文件，fastq.gz文件约为330G

#1.2下载人类基因组文件和人类基因组注释文件

wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_42/GRCh38.p13.genome.fa.gz
wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_42/gencode.v42.annotation.gtf.gz

gunzip gencode.v42.annotation.gtf.gz

gunzip GRCh38.p13.genome.fa.gz

#1.3 质控
##1.3.1首先将单端测序（SE）和双端测序（PE）数据分开
mkdir 01.SE
mkdir 02.PE
mv *_1.fastq.gz ./02.PE/
mv *_2.fastq.gz ./02.PE/
mv *.fastq.gz ./01.SE/
##1.3.2使用fastp进行质控
cd 	./01.SE/
for i in *.fastq.gz
do
i=${i%.fastq.gz*}
fastp -i ${i}.fastq.gz -o ${i}.clean.fastq.gz -w 8 --html ${i}.html --json ${i}.json
done

cd 	./02.PE/
for i in *_1.fastq.gz
do
i=${i%_1.fastq.gz*}
fastp -i ${i}_1.fastq.gz -I ${i}_2.fastq.gz -o ${i}_1.clean.fastq.gz -O ${i}_2.clean.fastq.gz -w 8 --html ${i}.html --json ${i}.json
done

#1.4 histat比对
##1.4.1 建立索引
hisat2-build GRCh38.p13.genome.fa hisat2_index_GRCh38
##1.4.2 比对
###1.4.2.1 单端测序数据比对
cd 	./01.SE/
for i in *.clean.fastq.gz
do
i=${i%.clean.fastq.gz*}
hisat2 -p 8 -x /home/xjh3/data/ach/cqsw_test/02.ref/hisat2_index_GRCh38 -U ${i}.clean.fastq.gz -S ${i}.sam >${i}.mapping.log 2>&1 
samtools sort --threads 8 -o ${i}.sort.bam ${i}.sam 
rm ${i}.sam
done

for i in *.sort.bam
do
i=${i%.sort.bam*}
samtools flagstat ${i}.sort.bam --threads 12 > ${i}.txt
done

mkdir 03.mapping.log
cd 03.mapping.log
cp ../01.SE/*.txt ./



###1.4.2.1 双端测序数据比对
cd 	./02.PE/
for i in *_1.clean.fastq.gz
do
i=${i%_1.clean.fastq.gz*}
hisat2 -x /home/xjh3/data/ach/cqsw_test/02.ref/hisat2_index_GRCh38 -p 8 -1 ${i}_1.clean.fastq.gz -2 ${i}_2.clean.fastq.gz -S ${i}.sam > ${i}.mapping.log 2>&1 
samtools sort --threads 8 -o ${i}.sort.bam ${i}.sam 
rm ${i}.sam
done

for i in *.sort.bam
do
i=${i%.sort.bam*}
samtools flagstat ${i}.sort.bam --threads 12 > ${i}.txt
done

cd 03.mapping.log
cp ../02.PE/*.txt ./

##1.4.3 统计比对情况：

ls *txt|while read id
do(echo $id|awk -F "." '{print $1}'>>sample)
done
ls *txt|while read id
do(cat $id|cut -d " " -f 1|sed -n "1p"|awk '{printf "%'"'"'18.0f\n",$0}' >>totalreads)
done
ls *txt|while read id
do(cat $id|cut -d " " -f 1|sed -n "2p"|awk '{printf "%'"'"'18.0f\n",$0}'>>secondary)
done
ls *txt|while read id
do(cat $id|cut -d " " -f 4,5|sed -n '5p'|awk -F "(" '{print $2}' >>mapratio)
done
echo "sample    totalreads      secondary       mapratio" >mapinfo.txt
paste -d "\t" sample totalreads secondary mapratio >>mapinfo.txt
###比对率很高，均大于94%

#1.5 featurecount定量
##1.5.1 单端数据定量
/home/xjh3/data/qtf/lncrna/deseq/subread-2.0.3-Linux-x86_64/bin/featureCounts -T 8 \
-t exon -g gene_id \
-a /home/xjh3/data/ach/cqsw_test/02.ref/gencode.v42.annotation.gtf -o all_SE.id.count.txt *.sort.bam

cat all_SE.id.count.txt | cut -f1,7- > SE_counts.txt
 
##1.5.2 双端数据定量
/home/xjh3/data/qtf/lncrna/deseq/subread-2.0.3-Linux-x86_64/bin/featureCounts -T 8 \
-t exon -g gene_id -p \
-a /home/xjh3/data/ach/cqsw_test/02.ref/gencode.v42.annotation.gtf -o all_PE.id.count.txt *.sort.bam
 cat all_PE.id.count.txt | cut -f1,7- > PE_counts.txt
 
paste  SE_counts.txt PE_counts.txt >counts.txt
 
cut --complement -f 130 counts.txt > counts_1.txt

##1.5.2 备注，由于发现rsubread的featureCounts默认设置是countMultiMappingReads=TRUE，而linux下载的featureCounts的默认设置Multimapping reads not counted，本次未添加-M参数，是由于由于多重重叠 (multi-overlapping) read是指在meta-feature水平进行read count时与一个以上meta-feature重叠的read，或者在以feature水平进行read count时与多个feature重叠的read。是否对这些read进行count通常由实验类型决定。建议在RNA-seq实验中不计算与一个以上基因重叠的read或fragment，因为任何单个fragment都必须仅来源于一个靶基因，但无法确定真正靶基因的身份。（参考：https://zhuanlan.zhihu.com/p/617715377）
 

```

## 2.下游分析

```{r eval =F}
#利用DESeq2 R包进行差异分析
getwd()
library(DESeq2)
data <- read.table(file = 'C:/Users/Admin/Desktop/featurecounts/counts_1.txt',header = TRUE, row.names = 1,encoding="UTF-8")
sample<-names(data)
# 设置分组信息并构建dds对象，按照geuvadis_phenodata.table给的信息写
condition <- factor(c(rep("BA", 121), rep("Normal", 7), rep("BA", 50)),levels=c("BA","Normal"))
coldata <- data.frame(row.names = colnames(data), condition)
dds <- DESeqDataSetFromMatrix(countData=data, colData=coldata, design=~condition)
dds <- dds[ rowSums(counts(dds)) > 1, ]
# 使用DESeq函数估计离散度，然后差异分析获得res对象
dds_norm <- DESeq(dds)
res1 <- results(dds_norm,contrast=c("condition","BA","Normal")) #计算两组间差异

# 最后设定阈值，筛选差异基因，导出数据(全部数据。包括标准化后的count数)
res1 <- res1[order(res1$padj),] #排序
resdata1 <- merge(as.data.frame(res1), as.data.frame(counts(dds_norm, normalized=TRUE)), by="row.names", sort=FALSE)


deseq2_merge<-rbind(data.frame(group="BA_Normal",resdata1))
write.table(deseq2_merge,file="deseq2.table",row.names=F,quote=F,sep="\t")   #保存未过滤的所有基因结果，方便在后期筛选差异表达基因

diff_gene1 <- subset(res1, padj < 0.05 & (log2FoldChange > 2 | log2FoldChange < -2)) #筛选差异基因，也可在linux筛选

diff1 <- row.names(diff_gene1)   #差异基因ID

up_DEG1 <- subset(resdata1, padj < 0.05 & log2FoldChange > 2)

down_DEG1 <- subset(resdata1, padj < 0.05 & log2FoldChange < -2)

diff_merge<-rbind(data.frame(group="BA_Normal_up",up_DEG1),data.frame(group="BA_Normal_down",down_DEG1))
write.table(diff_merge,file="diff.table",row.names=F,quote=F,sep="\t")
```

## 3.琴型图展示BA和Normal之间的差异

```R eval =F
#1.参考数据发表文章：Luo Z, Shivakumar P, Mourya R, Gutta S et al. Gene Expression Signatures Associated With Survival Times of Pediatric Patients With Biliary Atresia Identify Potential Therapeutic Agents. Gastroenterology 2019 Oct;157(4):1138-1152.e14.的Figure 2，原文献使用柱状图展示BA和normal之间的基因表达量差异，之前由于是单端数据使用的RPKM值，由于后续数据库又补充了50个BA病例的转录组数据，本次打算基于FPKM值使用琴型图展示BA和Normal之间的差异。
#2.理由：FPKM与RPKM非常相似。RPKM是针对单端测序的RNA-seq而言的，其中每个reads对应于一个已测序的单个片段。FPKM用于双端测序的RNA-seq。使用双端测序RNA-seq，两个reads可以对应一个片段（Fragment）。RPKM和FPKM之间的唯一区别是FPKM考虑到两次reads可以映射到一个片段（因此它不会对该片段进行两次计数）。 即 单端测序：reads=fragments，双端测序：2 * reads≈fragments）--https://zhuanlan.zhihu.com/p/513391213

#3.计算178个样本的RPKM值
################################################################################################################################
rm(list = ls())

library(parallel) #并行计算  parApply parLapply parSaplly 
cl <- makeCluster(0.75*detectCores())  #设计启用计算机3/4的核
getwd()
##3.1利用GenomicFeatures包导入gtf处理
library(GenomicFeatures)

txdb <- makeTxDbFromGFF("./gencode.v42.annotation.gtf.gz",
                        format="gtf") 
exons_gene <- exonsBy(txdb, by = "gene") ###提取基因外显子
head(exons_gene)
##3.2计算总外显子长度：用reduce去除掉重叠冗余的部分，,width统计长度，最后计算总长度
exons_gene_lens <- parLapply(cl,exons_gene,function(x){sum(width(reduce(x)))}) 
exons_gene_lens[1:10]

##3.3转换为dataframe
geneid_efflen <- data.frame(geneid=names(exons_gene_lens),
                            efflen=as.numeric(exons_gene_lens))
write.csv(geneid_efflen,file = "./gene_length.csv",row.names = F)

data <- read.table(file = 'C:/Users/Admin/Desktop/featurecounts/counts_1.txt',header = TRUE, row.names = 1,encoding="UTF-8")

##3.4加载基因长度信息##############
library(data.table)

gene_length = fread("./gene_length.csv",data.table = F)

index = intersect(gene_length$geneid,row.names(data))

gene_length = gene_length[gene_length$geneid%in%index,]
data = data[gene_length$geneid,]

identical(gene_length$geneid,row.names(data))

countToFpkm <- function(counts, effLen)
{
    N <- sum(counts)
    exp( log(counts) + log(1e9) - log(effLen) - log(N) )
}
fpkms <- apply(data, 2, countToFpkm,effLen = gene_length$efflen)
fpkms.m<-data.frame(fpkms)
write.table(fpkms.m,file="FPKM.matrix",row.names=T,quote=F,sep="\t" )
################################################################################################################################

#4.获得ENSEMBLE对应的gene_name
gtf <- read_tsv("./gencode.v42.annotation.gtf/gencode.v42.annotation.gtf", comment="#", col_names=c('attributes'))
gtf$attributes %>% str_extract(., "gene_id \"[\\w|\\.]+") %>% str_remove(., "gene_id \"") -> gtf$gene_id

gtf$attributes %>% str_extract(., "gene_name \"[\\w|\\.]+") %>% str_remove(., "gene_name \"") -> gtf$gene_name

d <- gtf %>% select(gene_id,gene_name)
d_unique <- unique(d)

write.table(d_unique,file="gene_id_name.txt",row.names=F,quote=F,sep="\t" )

#5.选择deseq2差异分析最显著的4个上调基因（ENSG00000147257.16（GPC3）、ENSG00000167244.22（IGF2）、ENSG00000141736.14（ERBB2）、ENSG00000105894.13（PTN））和参考文献的一个差异显著上调基因（ENSG00000106483.12（SFRP4））琴型图展示（基于FPKM）
#5.1 绘制小提琴图

library(readr)
FPKM <- read_delim("FPKM.matrix", delim = "\t", escape_double = FALSE, trim_ws = TRUE)

selected_rows_1 <- FPKM[FPKM$gene_id == "ENSG00000147257.16", ]

selected_rows_2 <- FPKM[FPKM$gene_id == "ENSG00000167244.22", ]

selected_rows_3 <- FPKM[FPKM$gene_id == "ENSG00000141736.14", ]

selected_rows_4 <- FPKM[FPKM$gene_id == "ENSG00000105894.13", ]

selected_rows_5 <- FPKM[FPKM$gene_id == "ENSG00000106483.12", ]

merged_df <- rbind(selected_rows_1, selected_rows_2,selected_rows_3,selected_rows_4,selected_rows_5)

df_transposed <- t(merged_df)
df_transposed <- as.data.frame(df_transposed, stringsAsFactors = FALSE)
colnames(df_transposed) <- df_transposed[1, ]
df_transposed <- df_transposed[-1, ]
a<- rep(c("BA","Normal","BA"),times = c(121,7,50))
df_transposed$type <- a

ggplot(df_transposed, aes(type,ENSG00000147257.16)) + 
  geom_violin(trim = FALSE, position = position_dodge(0.9)) + 
  geom_boxplot(width = 0.2, position = position_dodge(0.9)) + 
  scale_fill_brewer(palette = "Set2") + 
  theme_bw(base_size = 15) 

is.data.frame(df_transposed)
head(df_transposed)

df_transposed$ENSG00000147257.16 <- as.numeric(df_transposed$ENSG00000147257.16)
df_transposed$ENSG00000167244.22 <- as.numeric(df_transposed$ENSG00000167244.22)
df_transposed$ENSG00000141736.14 <- as.numeric(df_transposed$ENSG00000141736.14)
df_transposed$ENSG00000105894.13 <- as.numeric(df_transposed$ENSG00000105894.13)
df_transposed$ENSG00000106483.12 <- as.numeric(df_transposed$ENSG00000106483.12)

library(ggpubr)

my_comparisons=list(c("BA","Normal"))

mytheme <- theme(axis.title.x=element_text(face="italic", color="black",size=20,family="serif"),
                 axis.title.y = element_text(colour="black",size=20,face = "bold",family="serif",),
                 axis.text.x = element_text(colour="black",size=16,face="bold",family="serif"),
                 axis.text.y = element_text(colour="black",size=16,face="bold",family="serif"))

p1 <- ggviolin(df_transposed, x="type", y="ENSG00000147257.16", fill = "type", 
         palette = c('lancet'))+ stat_compare_means(comparisons = my_comparisons,label = "p.signif")+labs(y = 'FPKM', x = "GPC3")+mytheme

p2 <- ggviolin(df_transposed, x="type", y="ENSG00000167244.22", fill = "type", 
         palette = c('lancet'))+ stat_compare_means(comparisons = my_comparisons,label = "p.signif")+labs(y = 'FPKM', x = "IGF2")+mytheme

p3 <- ggviolin(df_transposed, x="type", y="ENSG00000141736.14", fill = "type", 
         palette = c('lancet'))+ stat_compare_means(comparisons = my_comparisons,label = "p.signif")+labs(y = 'FPKM', x = "ERBB2")+mytheme

p4 <- ggviolin(df_transposed, x="type", y="ENSG00000105894.13", fill = "type", 
         palette = c('lancet'))+ stat_compare_means(comparisons = my_comparisons,label = "p.signif")+labs(y = 'FPKM', x = "PTN")+mytheme

p5 <- ggviolin(df_transposed, x="type", y="ENSG00000106483.12", fill = "type", 
         palette = c('lancet'))+ stat_compare_means(comparisons = my_comparisons,label = "p.signif")+labs(y = 'FPKM', x = "SFRP4")+mytheme


p_A <- p1+p2+p3+p4+p5

p_A
ggsave('BA_Normal_.jpg', p_A, device = "jpg", dpi = 300, 
       width=16, height=8, unit = "in")
```

